src/Raven.Server/Documents/AI/Embeddings/StreamedEmbeddingValue.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.IO;
+using Sparrow.Server;
+
+namespace Raven.Server.Documents.AI.Embeddings;
+
+public class StreamedEmbeddingValue : IEmbeddingValue
+{
+    private readonly Stream _stream;
+
+    public StreamedEmbeddingValue(Stream stream)
+    {
+        _stream = stream;
+    }
+
+    public ReadOnlySpan<byte> GetEmbedding()
+    {
+        throw new NotSupportedException($"Getting embedding value isn't supported. Use {nameof(ReadTo)}() instead");
```

Comment:
This creates a fragile design where clients can't rely on the interface contract. Should we either modify the interface to include the `ReadTo()` method or consider a different design that ensures all implementations support the core methods without throwing exceptions?

If you'd like, I can try to implement such a solution.

Reply:
This is server code and we're handling implementations via pattern matching.

Reply:
I agree that it was a hack. I ended up with adding `Stream GetEmbeddingStream()` to `IEmbeddingValue` interface